components/OAuth2.0/authorization.c

```diff
@@ -0,0 +1,423 @@
+                    AuthPrivateHandler.Status = LOGIN;
+                    break;
+                }
+
+                xSemaphoreGive((*AuthInterfaceHandler->IsServiceAuthorizedSemaphore));                              
+                AuthPrivateHandler.TokenLastUpdate = xTaskGetTickCount();                                           
+                AuthPrivateHandler.Status = CHECK_TIME;                                                            
+                break;
+            }
+        }   
+        vTaskDelay(pdMS_TO_TICKS(100));            // Task delay at the end of while(1) loop
+    }
+}
+
+/**
+ * @brief This function initiates the authorization process.
+ * @param InterfaceHandler as the handler
+ * @return true if task run to the end, false if any error occurs
+ */
+bool Oauth_TaskInit(OAuthInterfaceHandler_t *InterfaceHandler)
+{
+    AuthInterfaceHandler = InterfaceHandler;
+    AuthPrivateHandler.Status = INIT;
+    if (AuthInterfaceHandler->ConfigAddressInSpiffs != NULL &&
+        AuthInterfaceHandler->IsServiceAuthorizedSemaphore != NULL)
+    {
+        StaticTask_t *xTaskBuffer = (StaticTask_t *)malloc(sizeof(StaticTask_t));
+        StackType_t *xStack = (StackType_t *)malloc(SERVICE_TASK_STACK_SIZE * sizeof(StackType_t));                 
+        if (xTaskBuffer == NULL || xStack == NULL)
+        {
+            ESP_LOGE(TAG, "Memory allocation failed!\n");
+            free(xTaskBuffer);
+            free(xStack);
+            return false; // Exit with an error code
+        }
+        xTaskCreateStatic(
+            Oauth_MainTask,                         // Task function
+            "Oauth_MainTask",                       // Task name (for debugging)
+            SERVICE_TASK_STACK_SIZE,                // Stack size (in words)
+            NULL,                                   // Task parameters (passed to the task function)
+            tskIDLE_PRIORITY + SERVICE_PRIORITY,    // Task priority (adjust as needed)
+            xStack,                                 // Stack buffer
+            xTaskBuffer                             // Task control block
+        );
+        // Allocate buffer and Initialize the service API call //TODO change dynamic mem to static
+        AuthPrivateHandler.OAuthBuffer.MessageBuffer = (char *)malloc(SUPER_BUF * sizeof(char));    
+        AuthPrivateHandler.OAuthBuffer.ResponseReadyFlag = xSemaphoreCreateBinary();
+        APICallInit(&AuthPrivateHandler.OAuthBuffer, AuthInterfaceHandler->ClientConfig.base64Credintials);
+
+        ESP_LOGI(TAG, "App initiated successfully");
+    }
+    else
+    {
+        ESP_LOGE(TAG, "IntefaceHandler is is missing some pointers, can not run the app");
+        return false;
+    }
+    return true;
+}
+
+/**
+ * @brief This function returns the access token.
+ * @param OAuthInterfaceHandler as the handler
+ * @return the access token
+ */
+char* Oauth_GetAccessToken(void)
+{
+    return AuthPrivateHandler.token.AccessToken;
+}
```

Comment:
why you don't use queue?

Reply:
this function just reads the memory and returns the value. No wait or inter-thread communication is needed